core/state/state_object.go

```diff
@@ -313,25 +313,19 @@ func (s *stateObject) updateTrie() (Trie, error) {
 		return s.trie, nil
 	}
 	// Retrieve a pretecher populated trie, or fall back to the database
-	tr, err := s.getPrefetchedTrie()
-	switch {
-	case err != nil:
-		// Fetcher retrieval failed, something's very wrong, abort
-		s.db.setError(err)
-		return nil, err
-
-	case tr == nil:
+	tr := s.getPrefetchedTrie()
+	if tr == nil {
 		// Fetcher not running or empty trie, fallback to the database trie
-		tr, err = s.getTrie()
+		_, err := s.getTrie()
```

Comment:
it's wrong. 

originally `tr` will be assigned here and finally be returned by the function. 
now we just leave it as `nil` if the trie is not available in prefetcher

Reply:
Yeah.  just pushed a fix for that.